### PR TITLE
Bug 1210694 - [Stingray][fling-player][TV][2.5] Require control panel funcitons

### PR DESCRIPTION
**Pull request by Fischer-L**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c20c54b343641fe7bb031a923382e0ba92d19ac6

File: tv_apps/fling-player/js/fling_player.js [hunk 1]
```diff
@@ -164,79 +186,180 @@
...
-  proto._stopAutoSeek = function fp_stopAutoSeek(dir) {
+  proto._stopAutoSeek = function (dir) {
     this._autoSeekStartTime = null;
     this._autoSeekDirection = null;
   };
 
-  proto._autoSeek = function fp_autoSeek() {
+  /**
+   * This is to handle this case that user seeks on video by long pressing key.
+   * Ithe seeking policy would go based on duration of pressing
```
Comment (rexboy7) [inline tv_apps/fling-player/js/fling_player.js]: typo

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3e52928efc0826cd865b4fd7b899a242f07783f6

File: tv_apps/fling-player/js/fling_player.js [hunk 2]
```diff
@@ -100,14 +102,33 @@
...
+    switch (state) {
+      case 'playing':
+        this._playButton.textContent = 'Play';
+      break;
+
+      case 'paused':
+        this._playButton.textContent = 'Pause';
+      break;
+    }
+  };
```
Comment (evelynhung) [inline tv_apps/fling-player/js/fling_player.js]: The text to indicate state seems wrong to me. Although it's just for your development testing, it's better to make it in a correct state. Please set: 
playing --> Pause
paused --> Play

File: tv_apps/fling-player/js/fling_player.js [hunk 3]
```diff
@@ -164,79 +186,176 @@
...
       this._autoSeekStartTime = (new Date()).getTime();
       this._autoSeekDirection = dir;
       this._autoSeek();
     }
   };
 
-  proto._stopAutoSeek = function fp_stopAutoSeek(dir) {
+  proto._stopAutoSeek = function (dir) {
     this._autoSeekStartTime = null;
     this._autoSeekDirection = null;
```
Comment (evelynhung) [inline tv_apps/fling-player/js/fling_player.js]: Please do `clearTimout`

File: tv_apps/fling-player/js/fling_player.js [hunk 4]
```diff
@@ -164,79 +186,176 @@
...
   };
 
-  proto._autoSeek = function fp_autoSeek() {
+  /**
+   * This is to handle this case that user seeks on video by long pressing key.
+   * The seeking policy would go based on duration of pressing
+   */
+  proto._autoSeek = function () {
```
Comment (evelynhung) [inline tv_apps/fling-player/js/fling_player.js]: nit: `repeatedSeek`?

File: tv_apps/fling-player/js/fling_player.js [hunk 5]
```diff
@@ -164,79 +186,176 @@
...
+    this._player.play();
+    this.setPlayButtonState('playing');
+  };
+
+  proto.pause = function () {
+    this._player.pause();
+    this.setPlayButtonState('paused');
+  };
+
+  proto._startAutoUpdateControlPanel = function () {
```
Comment (evelynhung) [inline tv_apps/fling-player/js/fling_player.js]: nit: `updatePlayingControlPanel`

File: tv_apps/fling-player/js/fling_player.js [hunk 6]
```diff
@@ -164,79 +186,176 @@
...
+      break;
+
       case 'playing':
-        this.setLoading(false);
+        this.showLoading(false);
+        // TODO: Hide 'Starting video cast from ...'
         this._connector.reportStatus('buffered', data);
+
+        this._startAutoUpdateControlPanel();
         this.showControlPanel(true);
```
Comment (evelynhung) [inline tv_apps/fling-player/js/fling_player.js]: Let UI controlled by the same function. move `showControlPanel` to the function you updating control panel.

File: tv_apps/fling-player/js/fling_player.js [hunk 7]
```diff
@@ -164,79 +186,176 @@
...
+
+  proto._stopAutoUpdateControlPanel = function () {
+    clearTimeout(this._autoUpdateTimer);
+    this._autoUpdateTimer = null;
+  };
+
+  /**
+   * This is to keep auto updating the info and status on the control panel.
+   */
+  proto._autoUpdateControlPanel = function () {
```
Comment (evelynhung) [inline tv_apps/fling-player/js/fling_player.js]: nit: `periodicallyUpdateControlPanel`?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4d77343200abf023bf159a418a99de5796b54873

File: tv_apps/fling-player/js/fling_player.js [hunk 8]
```diff
@@ -164,79 +188,181 @@
...
+        this._updateControlPanelContinuously.bind(this)
+      );
+      this.showControlPanel(true);
     }
   };
 
-  proto._stopAutoSeek = function fp_stopAutoSeek(dir) {
-    this._autoSeekStartTime = null;
-    this._autoSeekDirection = null;
+  proto._stopUpdateControlPanelContinuously = function () {
```
Comment (evelynhung) [inline tv_apps/fling-player/js/fling_player.js]: nit: I still feel the function name is not good enough, but let's not worry about it now and move forward

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f287c2a75bc38873eb5f33bdac9daf61e8d0d283

**Pull request merged**